static/js/components/NicDeviceSelector.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useState,
+} from "react";
+import { Input, Select } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { fetchNetworks } from "../api/networks";
+import { NotificationHelper } from "../types/notification";
+import { queryKeys } from "../util/queryKeys";
+import { LxdNicDevice } from "../types/device";
+
+interface Props {
+  notify: NotificationHelper;
+  setNicDevice: Dispatch<SetStateAction<LxdNicDevice | null>>;
+}
+
+const NicDeviceSelector: FC<Props> = ({ notify, setNicDevice }) => {
+  const [nic, setNic] = useState<string>("");
+  const [nameInInstance, setNameInInstance] = useState<string>("");
```

Comment:
Same idea with state here: parent can pass in nicDevice and this component could become stateless and much simpler. I.e. the updateNic can become a one liner with spreading: setNicDevice({...nicDevice, network: e.target.value})